src/cli/app-instance.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Worker thread that runs an instance of a mocked local fulfillment app.
+ */
+import {parentPort, workerData} from 'worker_threads';
+import {smarthomeStub, extractStubs} from '../platform/stub-setup';
+import {AppStub} from '../platform/smart-home-app';
+import {createSimpleExecuteCommands} from '../platform/execute';
+import {IdentifyMessage, ExecuteMessage} from './command-processor';
+import * as fs from 'fs';
+import * as path from 'path';
+
+/**
+ * Override the constructor to capture the AppStub instance.
+ */
+let appStubInstance: AppStub | undefined = undefined;
+class CliAppStub extends AppStub {
+  constructor(version: string) {
+    super(version);
+    appStubInstance = this;
+  }
+}
+
+/**
+ * Inject the stubs into the worker thread context.
+ */
+smarthomeStub.App = CliAppStub;
+(global as any).smarthome = smarthomeStub;
+
+/**
+ * Validate the filepath.
+ */
+try {
+  // Strips the file extension off the filepath,
+  const modulePath = workerData.substring(0, workerData.lastIndexOf('.'));
+  fs.existsSync(path.relative('./', modulePath));
```

Comment:
Doesn't this return true/false rather than throw?

Reply:
https://nodejs.org/api/fs.html#fs_fs_existssync_path

Reply:
Changed to checking if false and throwing a new `Error`.